### PR TITLE
refactor: support multiple signature suites for VC and VP verification

### LINKED ISSUE #1602: VC/VP verification supports only one signature suite
Current code, supports only one verifierSuite for verification.
1. The consumer need to retrieve the proof, check the type and then need to figure out the verifierSuite. This will add extra hop at the consumer layer.
2. Also, this will not work when VC/VP has multiple proofs with different types. 

https://github.com/hyperledger/aries-framework-go/blob/c043ad2f2e922bb2605ce13b253920cd28d0cc1d/pkg/doc/verifiable/credential.go#L575

Ideally, the API should be able to take multiple verifierSuites and internally need to find teh suite based on proof type.

### PR DESCRIPTION
**Pull request by kdimak**
closes #1602

Signed-off-by: Dmitriy Kinoshenko <dkinoshenko@gmail.com>

### CODE REVIEW

File: pkg/doc/signature/verifier/verifier.go [hunk 1]
```diff
@@ -48,14 +48,14 @@ type keyResolver interface {
...
 // DocumentVerifier implements JSON LD document proof verification
 type DocumentVerifier struct {
-	signatureSuites []signatureSuite
+	signatureSuites []SignatureSuite
 	pkResolver      keyResolver
 }
 
 // New returns new instance of document verifier
-func New(resolver keyResolver, mainSuite signatureSuite, extraSuites ...signatureSuite) *DocumentVerifier {
+func New(resolver keyResolver, mainSuite SignatureSuite, extraSuites ...SignatureSuite) *DocumentVerifier {
```
Comment (rolsonquadras) [inline pkg/doc/signature/verifier/verifier.go:56]: Why do we have distinction between mainSuite and extraSuites? And how should the consumer decide between main and extra ?

File: pkg/doc/verifiable/linked_data_proof_test.go [hunk 2]
```diff
@@ -78,21 +84,91 @@ func TestLinkedDataProofSignerAndVerifier(t *testing.T) {
...
+				return &verifier.PublicKey{
+					Type:  "EcdsaSecp256k1VerificationKey2019",
+					Value: pubKeyBytes,
+					JWK: &jose.JWK{
+						JSONWebKey: gojose.JSONWebKey{
+							Algorithm: "ES256K",
+						},
+						Crv: "secp256k1",
+						Kty: "EC",
+					},
```
Comment (rolsonquadras) [inline pkg/doc/verifiable/linked_data_proof_test.go:142]: The [spec](https://w3c-ccg.github.io/ld-cryptosuite-registry/#ecdsa-secp256k1) doesn't specify JWK for `EcdsaSecp256k1VerificationKey2019` key type. Is this a dummy data for unit test or does the code really expect the DIDDoc to have JWK value for this key type?

File: pkg/doc/signature/verifier/verifier.go (hunk 1 above)
Comment (kdimak) [reply pkg/doc/signature/verifier/verifier.go:56]: I used this approach to force a definition at least one signature suite. There is no difference between the main and extra signature suites.

File: pkg/doc/verifiable/linked_data_proof_test.go (hunk 2 above)
Comment (kdimak) [reply pkg/doc/verifiable/linked_data_proof_test.go:142]: At https://w3c-ccg.github.io/lds-ecdsa-secp256k1-2019/, you can find an example with the JWK. `publicKeyHex` should be supported in the `EcdsaSecp256k1Signature2019` suite, I will create an issue for that.

Comment (kdimak) [reply pkg/doc/verifiable/linked_data_proof_test.go:142]: https://github.com/hyperledger/aries-framework-go/issues/1609

File: pkg/doc/signature/verifier/verifier.go (hunk 1 above)
Comment (sandrask) [reply pkg/doc/signature/verifier/verifier.go:56]: I agree that we shouldn't have mainSuite and extra suites. You can provide a comment saying that at least one suite is required and check for suites length in New().

Comment (troyronda) [reply pkg/doc/signature/verifier/verifier.go:56]: the terminology "main" and "extra" is likely to be confusing.

Comment (kdimak) [reply pkg/doc/signature/verifier/verifier.go:56]: @rolsonquadras @sandrask @troyronda 
Created https://github.com/hyperledger/aries-framework-go/issues/1616

**Pull request merged**